Scripts/timer_canvas.gd

```diff
@@ -1,9 +1,24 @@
 extends CanvasLayer
 
 @onready var timer_text: Label = $timer_text
-@export var timer: float = 120.0
+@export var time_limit: float = 120.0
+var timer: float
+var continueCountdown: bool = true
+
+# Sets the value of timer to be equal to time_limit on start.
+func _ready() -> void:
+	timer = time_limit
 
 # Called every frame. 'delta' is the elapsed time since the previous frame.
 func _process(delta: float) -> void:
-	timer -= delta
-	timer_text.text = str(timer)
+	if (continueCountdown):
```

Comment:
this should ideally be an early exit condition, this prevents excessive indentation.
Essentially if everything following an if statement would only run if it is true then flip it and just return, makes the code easier to read.

if not contineCoiuntdown: return